### ISSUE TITLE
Exception on PeerGroup shutdown

### ISSUE THREAD

**Issue by teaforthecat** [onyx-platform/onyx#821]
```
1. Caused by java.lang.ClassCastException
   com.codahale.metrics.MetricRegistry cannot be cast to clojure.lang.IFn

                  core.clj:   48  metrics.core/remove-metrics
                  core.clj:   41  metrics.core/remove-metrics
                  core.clj:   45  metrics.core/remove-metrics
                  core.clj:   41  metrics.core/remove-metrics
    metrics_monitoring.clj:  152  onyx.monitoring.metrics-monitoring.Monitoring/stop
```

It looks like when metrics.core/remove-metrics takes a single argument, it is expected to be a function "pred".

**Comment by teaforthecat** [comment]
It looks like metrics.core/remove-metric has changed in the newer version of 2.10.0, was the intention to upgrade the metrics-clojure dependency perhaps?

**Comment by teaforthecat** [comment]
Also note that this will only be triggered if there are metrics in the `metrics.core/default-registry` .

**Comment by lbradstreet** [comment]
Thanks. I'm not sure why we don't see any exceptions on our end. You're right that the usage is wrong.

Could you please confirm that it's fixed on `0.11.2-20171101.172315-2`?

**Comment by lbradstreet** [comment]
Fixed.
